### ISSUE TITLE
Collision detection sensor possible?

### ISSUE THREAD

**Issue by MrAnDrs** [14yannick/esphome-hcpbridge#9]
Is there any chance to add a collision detected / error sensor? 

I previously had a Homee "Hörmann" Cube and it had a sensor for "Problem occured". I also saw it here: https://github.com/stephan192/hoermann_door 
but my coding skills are quite limited. 

Thanks!

**Comment by MrAnDrs** [comment]
Hi, yes I do have an E4. Sorry my bad about linking to Stephan’s project. Obviously I missed that part. Thanks for sharing these information with me, hoping you will find something. If I can somehow help, let me know. My wife is good in reproducing the error, lately. That’s why I was asking 🙈.

 I recently switched from the mqtt implementation to yours and love the status values with the explicit venting status. What I did not find yet is the percentage of the opened door. In the mqtt one the door was just shown as opened with around 6% opened when it was venting and when it collided with a car and did a safety opening again it also showed opened with around 4% open. So not very easy to divide.

**Comment by 14yannick** [comment]
I haven't checked yet if the error code is also transmitted in the status broadcast. But as written above by setting the value 06 in menu 30 of your motor the `Relay state` binyry sensor will be true as soon a malfunction is detected. to get the current position in percentage you could add a template sensor like this.
```YAML
  - platform: template
    id: "gd_percent"
    name: "Garage Door percentage"
    unit_of_measurement: "%"
    state_class: measurement
    icon: "mdi:percent" 
    accuracy_decimals: 0
    lambda: |-
      return id(garagedoor_cover).position * 100;
    update_interval: 5s
```
tell me if it shows the expected values for you

**Comment by MrAnDrs** [comment]
Hi, 
the sensor works as expected, thanks for the code! I modified a bit and sticked to your yaml configuration and moved the sensor name to the top "configuration" area as a variable.
It "just" floods the ESP log with the 5 seconds update. But dont know if this is harmful?!?

I also changed the option in the menu 30 - but no chance yet to test. If that is working fine, for me this "workaround" is sufficient enough, as I dont need that output for something else. Thanks for explaining, it was very helpful and put me on the right track!
